kymatio/scattering1d/backend/numpy_backend.py

```diff
@@ -0,0 +1,213 @@
+
+def pad(x, pad_left=0, pad_right=0, to_complex=True):
+    """Pad real 1D tensors and map to complex
+
+    Padding which allows to simultaneously pad in a reflection fashion and map
+    to complex if necessary.
+
+    Parameters
+    ----------
+    x : tensor
+        Three-dimensional input tensor with the third axis being the one to
+        be padded.
+    pad_left : int
+        Amount to add on the left of the tensor (at the beginning of the
+        temporal axis).
+    pad_right : int
+        amount to add on the right of the tensor (at the end of the temporal
+        axis).
+    to_complex : boolean, optional
+        Whether to map the resulting padded tensor to a complex type (seen
+        as a real number). Defaults to True.
+
+    Returns
+    -------
+    output : tensor
+        A padded signal, possibly transformed into a four-dimensional tensor
+        with the last axis of size 2 if to_complex is True (this axis
+        corresponds to the real and imaginary parts).
+    """
+    output = pad_1d(x, pad_left, pad_right, mode='reflect')
+    return output
+
+def unpad(x, i0, i1):
+    """Unpad real 1D tensor
+
+    Slices the input tensor at indices between i0 and i1 along the last axis.
+
+    Parameters
+    ----------
+    x : tensor
+        Input tensor with least one axis.
+    i0 : int
+        Start of original signal before padding.
+    i1 : int
+        End of original signal before padding.
+
+    Returns
+    -------
+    x_unpadded : tensor
+        The tensor x[..., i0:i1].
+    """
+    return x[..., i0:i1]
+
+def real(x):
+    """Real part of complex tensor
+
+    Takes the real part of a complex tensor, where the last axis corresponds
+    to the real and imaginary parts.
+
+    Parameters
+    ----------
+    x : tensor
+        A complex tensor (that is, whose last dimension is equal to 2).
+
+    Returns
+    -------
+    x_real : tensor
+        The tensor x[..., 0] which is interpreted as the real part of x.
+    """
+    return np.real(x)
+
+def fft1d_c2c(x):
+    """Compute the 1D FFT of a complex signal
+
+    Input
+    -----
+    x : tensor
+        A tensor of size (..., T, 2), where x[..., 0] is the real part and
+        x[..., 1] is the imaginary part.
+
+    Returns
+    -------
+    x_f : tensor
+        A tensor of the same size as x containing its Fourier transform in the
+        standard PyTorch FFT ordering.
+    """
+    return np.fft.fft(x)
+
+def ifft1d_c2c(x):
+    """Compute the normalized 1D inverse FFT of a complex signal
+
+    Input
+    -----
+    x_f : tensor
+        A tensor of size (..., T, 2), where x_f[..., 0] is the real part and
+        x[..., 1] is the imaginary part. The frequencies are assumed to be in
+        the standard PyTorch FFT ordering.
+
+    Returns
+    -------
+    x : tensor
+        A tensor of the same size of x_f containing the normalized inverse
+        Fourier transform of x_f.
+    """
+    return np.fft.ifft(x)
+
+def finalize(s0, s1, s2):
```

Comment:
I think `finalize` seems to make sense... we're 'finalizing' the scattering transform by stacking the coefficients together. Thoughts @edouardoyallon ?

Reply:
I'm pros finalize too

Reply:
Because, again, it could mean *anything*. It could be applying the lowpass filter to the wavelet modulus coefficients. It could be computing the log (which is a common postprocessing step in 1D scattering for audio). Why use a generic term when a specific one will do?